Documentation/WritingReST/Reference/Code/Confval.rst

```diff
@@ -28,49 +27,175 @@ Each configuration value name may only be used once. In large references
 with different contexts you can define individual configuration schemas for
 each context.
 
+..  contents:: Table of contents
+
+..  _rest-confval-examples:
+
 Examples
 ========
 
 Required configuration value
 ----------------------------
 
 ..  confval:: label
-
-    :Required: true
+    :name: some-unique-label
+    :required: true
     :type: string or LLL reference
-    :Scope: Display
-    :Path: $GLOBALS > TCA > [table] > columns > [field]
 
     The name of the field as shown in the form.
 
 ..  code-block:: rst
 
     ..  confval:: label
-
-        :Required: true
+        :name: some-unique-label
+        :required: true
         :type: string or LLL reference
-        :Scope: Display
-        :Path: $GLOBALS > TCA > [table] > columns > [field]
 
         The name of the field as shown in the form.
 
-Configuration value with default value
---------------------------------------
+Example: Configuration value with default value adn custom parameter
+--------------------------------------------------------------------
 
 ..  confval:: fileCreateMask
-
+    :name: some-unique-fileCreateMask
     :type: text
-    :Default: 0664
-    :Path: $GLOBALS > TYPO3_CONF_VARS > SYS
+    :default: 0664
+    :Path: :php:`$GLOBALS['TYPO3_CONF_VARS']['SYS']['fileCreateMask']`
 
     File mode mask for Unix file systems (when files are uploaded/created).
 
 ..  code-block:: rst
 
     ..  confval:: fileCreateMask
-
+        :name: some-unique-fileCreateMask
         :type: text
-        :Default: 0664
-        :Path: $GLOBALS > TYPO3_CONF_VARS > SYS
+        :default: 0664
+        :Path: :php:`$GLOBALS['TYPO3_CONF_VARS']['SYS']['fileCreateMask']`
 
         File mode mask for Unix file systems (when files are uploaded/created).
+
+Confval directive API
+=====================
+
+Each confval must have at least a title. If that title is not unique within
+the manual the confval must also have the `:name:` attribute, followed by a
+unique name. Names are case-insensitive and convert all special signs into a dash.
+
+..  code-block:: rst
+
+    ..  confval:: [title]
+        :name: [unique-name]
+
+There are several reserved attributes:
+
+`:type:`
+    The type of the configuration value.
+`:default:`
+    The default value
+`:required:`
+    Is the configuration value required.
+`:name:`
+    The unique identifier, reserved internally by reStructuredText.
+`:class:`
```

Comment:
I havn't found `class` here: https://github.com/sphinx-toolbox/sphinx-toolbox/blob/master/sphinx_toolbox/confval.py. Do we have a special implementation for TYPO3 docs?

Reply:
class and name are defined and reserved for any directive.